contrib/packaging/docker/Dockerfile.runtime

```diff
@@ -43,13 +43,13 @@ RUN go get -d github.com/google/gops && \
 # LLVM, bpftool, iproute2 and loopback for runtime image (cilium/packaging repo)
 #
 FROM runtime-base as tools
-ARG ARCH=amd64
 WORKDIR /tmp
 # when updating this version, also run `images/scripts/update-cni-version.sh <v>`
 # to update images/runtime/cni-version.sh
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
       curl ca-certificates xz-utils binutils && \
+      ARCH=$(uname -m) && [ "$ARCH" != "aarch64" ] || ARCH="arm64" && [ "$ARCH" != "x86_64" ] || ARCH="amd64" && \
```

Comment:
What does this do exactly? The chaining together of the variable configurations and the tests of some options (for example `ARCH="arm64" && [ "$ARCH != "x86_64 ]` which seems like it would always return true) seems confusing to me.

Reply:
(I note this is not the only place in this PR with that logic. Maybe it's a common known pattern I'm unaware of, but that doesn't make it any less confusing ;) )

Reply:
`uname -m` returns `aarch64` for ARM64 and `x86_64` for AMD64, respectively, while `ARCH` value needs to be `arm64` or `amd64`, respectively. The pattern of interest is `[ "$ARCH" != "aarch64" ] || ARCH="arm64"` and `[ "$ARCH" != "x86_64" ] || ARCH="amd64"`, while the `&&` is used to chain together multiple (successful) commands in a single dockerfile `RUN` command, typical in Dockerfiles to avoid building unnecessary filesystem layers and to be able to clean up temp files in the end of the `&&` chain.

`[ "$ARCH" != "aarch64" ] || ARCH="arm64"` executes the second part after `||` only if the first part before it is "untrue". So ARCH is set to "arm64" only if ARCH is "aarch64". Same for "x86_64" and "amd64".